appeals/web/src/server/appeals/appeal-details/costs/costs.controller.js

```diff
@@ -297,20 +328,74 @@ export const postDeleteCostsDocument = async (request, response) => {
 	const {
 		currentAppeal,
 		currentFolder,
-		params: { costsApplicant }
+		params: { costsCategory }
 	} = request;
 
 	if (!currentAppeal) {
 		return response.status(404).render('app/404');
 	}
 	if (!currentFolder) {
-		return response.status(404).render('app/500');
+		return response.status(404).render('app/404');
 	}
 
 	postDocumentDelete(
 		request,
 		response,
 		`/appeals-service/appeal-details/${request.params.appealId}`,
-		`/appeals-service/appeal-details/${currentAppeal.appealId}/costs/${costsApplicant}/upload-documents/${currentFolder?.id}/{{documentId}}`
+		`/appeals-service/appeal-details/${currentAppeal.appealId}/costs/${costsCategory}/upload-documents/${currentFolder?.id}/{{documentId}}`
+	);
+};
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const renderDecisionCheckAndConfirm = async (request, response) => {
+	const { errors, currentAppeal, currentFolder } = request;
+
+	if (!currentAppeal) {
+		return response.status(404).render('app/404');
+	}
+	if (!currentFolder) {
+		return response.status(404).render('app/404');
+	}
+
+	const mappedPageContent = decisionCheckAndConfirmPage(request.currentAppeal, currentFolder);
+
+	return response.render('patterns/check-and-confirm-page.pattern.njk', {
+		pageContent: mappedPageContent,
+		errors
+	});
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const getDecisionCheckAndConfirm = async (request, response) => {
+	renderDecisionCheckAndConfirm(request, response);
+};
+
+/** @type {import('@pins/express').RequestHandler<Response>} */
+export const postDecisionCheckAndConfirm = async (request, response) => {
+	const { session, errors, currentAppeal, currentFolder } = request;
+
+	if (!currentAppeal) {
+		return response.status(404).render('app/404');
+	}
+	if (!currentFolder) {
+		return response.status(404).render('app/404');
+	}
+
+	if (errors) {
+		return renderDecisionCheckAndConfirm(request, response);
+	}
+
+	addNotificationBannerToSession(
+		session,
+		'costsDocumentAdded',
+		currentAppeal.appealId,
+		'',
```

Comment:
Won't this leave a blank notification banner?

Reply:
The `text` argument overrides the `html` argument if `text` is non-zero length and `html` is an empty string - although it's not very clear from the code (see https://github.com/Planning-Inspectorate/appeals-back-office/blob/main/appeals/web/src/server/lib/mappers/notification-banners.mapper.js#L199). Maybe something we can work on as an improvement later!